From 5d0f6ea4704677f017597763e8ecf7431a2e7042 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 31 May 2021 10:14:28 +0300
Subject: [PATCH] [SW-0] gaudi2: change sram address for onix

For H6 FPGA onix, SRAM is mapped to a region after HBM, due to issue
related to Xilinx split bar. For this, change the SRAM base for
onix target accordingly.

Change-Id: I22798c5c4fe1eb9d9c86424a8cf32aec43a27579
---

diff --git a/Makefile b/Makefile
index 465b8a8..7600512 100644
--- a/Makefile
+++ b/Makefile
@@ -17,6 +17,9 @@
 ifeq ($(HL_TARGET),pldm)
 	CFLAGS += -DTARGET_IS_PLDM
 endif
+ifeq ($(HL_TARGET),onix)
+	CFLAGS += -DTARGET_IS_ONIX
+endif
 
 ifeq ($(FAKE_SECURITY),ON)
 	CFLAGS += -DFAKE_SECURITY
diff --git a/gaudi2/common.h b/gaudi2/common.h
index bcff904..b1b4374 100644
--- a/gaudi2/common.h
+++ b/gaudi2/common.h
@@ -9,6 +9,14 @@
 #include <gaudi2_blocks_linux_driver.h>
 #include <bitops.h>
 
+#ifdef TARGET_IS_ONIX
+#define PPBOOT_TEXT_BASE		0x8200000000
+#define SDRAM_BASE			0x8200000000ull
+#else
+#define PPBOOT_TEXT_BASE		0x7FFD000000
+#define SDRAM_BASE			0x7FFD000000ull
+#endif
+
 #define HL_PCIE_LINK_RETRIES             50
 #define HL_FW_BOOT_PTRS_BASE		0x80000
 #define HL_SPI_LEGACY_PREBOOT_OFF	0xC0000
@@ -16,7 +24,6 @@
 #define PPBOOT_SIZE			0x5000		/* 20KB */
 #define PREBOOT_SIZE			0x10000		/* 64KB */
 #define SRAM_SIZE			0x3000000	/* 48MB */
-#define PPBOOT_TEXT_BASE		0x7FFD000000
 #define PREBOOT_TEXT_BASE		(PPBOOT_TEXT_BASE + PPBOOT_SIZE)
 
 #define CFG_REGION_SIZE			0xC000000
@@ -24,7 +31,6 @@
 
 #define HL_SPI_BASE			0x3FF6000000ull
 #define HL_CFG_BASE			0x3FF8000000ull
-#define SDRAM_BASE			0x7FFD000000ull
 #define STM_FLASH_BASE			0x1000007FF4000000ull
 #define BAR0_RSRVD_BASE			0x1000FFFFFC000000ull
 #define DRAM_BASE			0x1001000000000000ull
